contracts/Strategy.sol

```diff
@@ -515,13 +601,15 @@ contract Strategy is BaseStrategy {
             return _amtInWei;
         }
 
-        uint256 price = uint256(chainlinkWantToETHPriceFeed.latestAnswer());
-        return _amtInWei.mul(WAD).div(price);
+        uint256 price = uint256(chainlinkWantUnderlyingTokenToETHPriceFeed.latestAnswer());
+        //TODO: check if this formula is working properly
+        return _amtInWei.mul(EXCHANGE_RATE_PRECISION).div(price).mul(wantAsVault.pricePerShare());
```

Comment:
@newmickymousse usually you want to place the division at the end to avoid missing precision. In addition, is this only supporting meta vaults of other yvTokens (wantAsVault) and not regular tokens?

Reply:
Exactly.. it's just for our yvTokens.. should we do it for regular tokens as well?

Reply:
Discussed this on telegram. There is going to be another branch for non-yvtokens.